### PR TITLE
[risk=no] [RW-6420] Survey Version table react component

### PR DESCRIPTION
**Pull request by SrushtiGangireddy**
Creating survey version table react component.

### CODE REVIEW

File: public-ui/src/app/data-browser/components/survey-version-table/survey-version-table-react.component.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    ElementRef,
+    EventEmitter,
+    Input,
+    Output,
+    ViewChild,
+    ViewEncapsulation
+} from '@angular/core';
+import * as React from 'react';
+import { FunctionComponent } from 'react';
+import * as ReactDOM from 'react-dom';
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/components/survey-version-table/survey-version-table-react.component.tsx]: Nit: remove unused imports

File: public-ui/src/app/data-browser/components/survey-version-table/survey-version-table-react.component.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+const containerElementName = 'root';
+
+interface Props {
+    surveys: any;
+}
+
+export class SurveyVersionTableReactComponent extends React.Component<Props, {}> {
+constructor(props: Props) {
+    super(props);
+    this.processSurveys();
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/components/survey-version-table/survey-version-table-react.component.tsx:34]: This should be called in `componentDidMount` (see [React docs](https://reactjs.org/docs/react-component.html#constructor))

File: public-ui/src/app/data-browser/components/survey-version-table/survey-version-table-react.component.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            return -1;
+        }
+        if (a1.monthNum.split('/')[0] > a2.monthNum.split('/')[0]) {
+            return 1;
+        }
+        return 0;
+    });
+this.props.surveys.forEach((survey, i) => {
+    survey['pdfLink'] = '/assets/surveys/' + survey.monthName.replace('/', '_') + '_COPE_COVID_English_Explorer.pdf';
+});
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/components/survey-version-table/survey-version-table-react.component.tsx]: While this works, it is not recommended to modify a component's own props (see [React docs](https://reactjs.org/docs/components-and-props.html#props-are-read-only)). In this case, I think the best way is to copy the original array to a const, then store it in state after modifying:
```suggestion
    const sortedSurveys = [...this.props.surveys].sort((a1, a2) => {
        if (a1.monthNum.split('/')[0] < a2.monthNum.split('/')[0]) {
            return -1;
        }
        if (a1.monthNum.split('/')[0] > a2.monthNum.split('/')[0]) {
            return 1;
        }
        return 0;
    });
sortedSurveys.forEach((survey, i) => {
    survey['pdfLink'] = '/assets/surveys/' + survey.monthName.replace('/', '_') + '_COPE_COVID_English_Explorer.pdf';
});
this.setState({surveys: sortedSurveys});
```

File: public-ui/src/app/data-browser/components/survey-version-table/survey-version-table-react.component.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            <div className='version-box'>
+                <div className='version-box-header'>
+                <div className='version-box-item'>Month</div>
+                <div className='version-box-item'>Participants</div>
+                <div className='version-box-item'>Number of Questions</div>
+                <div className='version-box-item'>Download PDF</div>
+            </div>
+            <div className='version-box-body'>
+            {
+                this.props.surveys.map((survey, index) => {
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/components/survey-version-table/survey-version-table-react.component.tsx]: if the above changes are made, we'll need to add condition here to only render after the surveys are set:
```suggestion
                !!this.state.surveys && this.state.surveys.map((survey, index) => {
```

Comment (dolbeew) [reply public-ui/src/app/data-browser/components/survey-version-table/survey-version-table-react.component.tsx]: Nit: remove unused `index`

File: public-ui/src/app/data-browser/components/survey-version-table/survey-version-table-react.component.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    this.props.surveys.sort((a1, a2) => {
+        if (a1.monthNum.split('/')[0] < a2.monthNum.split('/')[0]) {
+            return -1;
+        }
+        if (a1.monthNum.split('/')[0] > a2.monthNum.split('/')[0]) {
+            return 1;
+        }
+        return 0;
+    });
+this.props.surveys.forEach((survey, i) => {
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/components/survey-version-table/survey-version-table-react.component.tsx]: Nit: remove unused `i`

File: public-ui/src/app/data-browser/components/survey-version-table/survey-version-table-react.component.tsx (hunk 2 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/data-browser/components/survey-version-table/survey-version-table-react.component.tsx:34]: Moved this thanks.

File: public-ui/src/app/data-browser/components/survey-version-table/survey-version-table-react.component.tsx (hunk 3 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/data-browser/components/survey-version-table/survey-version-table-react.component.tsx]: Thanks good to know the practice.

**Review by dolbeew — APPROVED** [review]

**Pull request merged**